### DISCUSSION TITLE
How nuxt3 aggregates management api?

### DISCUSSION THREAD

**Discussion by balck-paint** [nuxt/nuxt#16643] [Questions]
**What can I do to aggregate and manage my api interface? I checked the official website documentation, but unfortunately there is no example of this. In fact, I don't know the fetch library very well, I can only write the following code based on the experience of using axios**
```
~/api/request.ts

import type { HTTPMethod } from 'h3'

const baseUrl = 'http://localhost:8100'

const errorResponse: ApiResponse = {
  success: false,
  code: 0,
  message: '',
  data: null,
}

const req = async (url: string, method: HTTPMethod, params = {}): Promise<ApiResponse> => {
  try {
    const token = useCookie('token')
    const res = await $fetch(baseUrl + url, {
      headers: {
        'Content-Type': 'application/json',
        'Authorization': token.value,
      },
      method,
      params,
    })
    return res as any
  }
  catch (error: any) {
    errorResponse.message = error
    return errorResponse
  }
}

export default req
```

```
~/api/auth.ts
import http from '~/api/request'

export function useAuthApi() {
  return {
    register(params: any): Promise<ApiResponse> {
      return http('/auth/auth-member/register', 'POST', params)
    },
    login(params: any): Promise<ApiResponse> {
      return http('/auth/auth-member/login', 'POST', params)
    },
    logout(): Promise<ApiResponse> {
      return http('/auth/auth-member/logout', 'DELETE')
    },
    getInfo(): Promise<ApiResponse> {
      return http('/auth/auth-member/info', 'GET')
    },
    getSmsCode(params: any): Promise<ApiResponse> {
      return http('/auth/auth-member/register', 'GET', params)
    },
  }
}
```
**When I try to access the backend interface, I get a cros error**
```
//index.vue
<script lang="ts" setup>
import { useAuthApi } from '~/api/auth'

definePageMeta({
  layout: 'default',
})

const { data: res } = await useAuthApi().getInfo()
console.log(res)
</script>
```
![image](https://user-images.githubusercontent.com/81019019/186630160-830bb1f6-ae7b-4982-b987-30ef437636c5.png)

**When I tried to imitate vite for cross-domain configuration, the result was...**

```
//nuxt.config.ts
 vite: {
    server: {
      proxy: {
        '/api': {
          target: 'http://localhost:8100',
          ws: true,
          changeOrigin: true,
          rewrite: path => path.replace(/^\/api/, ''),
        },
      },
    },
  },
```
![image](https://user-images.githubusercontent.com/81019019/186630961-a2c6620b-dc3d-44d6-9af6-1e2ae7d3b02b.png)

![image](https://user-images.githubusercontent.com/81019019/186631192-2d67f48d-9301-4fde-8f6f-e2de2c3f67d0.png)

**And why is the access path in the background a ts file! !
Seriously, I'm a little bit broken**
**Hope to get an answer**

**Comment by HomWang** [comment]
Please seting `MIME.type`

**Reply by balck-paint** [reply]
Thanks for your reply, can you elaborate?

**Reply by HomWang** [reply]
Add MIME.types type .mjs

**Comment by amihhs** [comment]
CORS  question, Maybe this will help us a little. [https://github.com/unjs/nitro/issues/113](https://github.com/unjs/nitro/issues/113)
